### PR TITLE
Fix: Add array access to CommonObject

### PR DESCRIPTION
**Pull request by mdeweerd**
# Fix: Add array access to CommonObject

Phan detects array access to subclasses of CommonObject (for instance: Societe). This adds array access to properties of CommonObjects.

### CODE REVIEW

File: htdocs/core/class/commonobject.class.php [hunk 1]
```diff
@@ -10714,4 +10714,80 @@ public function deleteEcmFiles($mode = 0)
...
+		}
+	}
+
+	/**
+	 * Provide list of available properties on this object
+	 *
+	 * (Not part of the ArrayAccess interface, provided for convenience.).
+	 * @return string[]
+	 */
+	public function getKeys()
```
Comment (eldy) [inline htdocs/core/class/commonobject.class.php:10789]: What is goal of this method ?

File: htdocs/core/class/commonobject.class.php [hunk 2]
```diff
@@ -43,7 +43,7 @@
 /**
  *	Parent class of all other business classes (invoices, contracts, proposals, orders, ...)
  */
-abstract class CommonObject
+abstract class CommonObject implements ArrayAccess
```
[the remark this answers is not in the document]
Comment (mdeweerd) [reply htdocs/core/class/commonobject.class.php:46]: The goal is not to detect bad properties.
DolDeprecatedHandler can handle that.

See comment on PR for the reason.

File: htdocs/core/class/commonobject.class.php (hunk 1 above)
Comment (mdeweerd) [reply htdocs/core/class/commonobject.class.php:10789]: See comment on PR for the reason.

**Comment by mdeweerd** [comment]
ArrayAccess is the standard interface to make an object look like an array.

The are the locations where a cCommonObject is accessed as an array.
```
htdocs\core\modules\project\doc\doc_generic_project_odt.modules.php:352 PhanTypeArraySuspicious Suspicious array access to $tasktime of type object
htdocs\core\modules\project\doc\doc_generic_project_odt.modules.php:353 PhanTypeArraySuspicious Suspicious array access to $tasktime of type object
htdocs\core\modules\project\doc\doc_generic_project_odt.modules.php:354 PhanTypeArraySuspicious Suspicious array access to $tasktime of type object
htdocs\core\modules\project\doc\doc_generic_project_odt.modules.php:355 PhanTypeArraySuspicious Suspicious array access to $tasktime of type object
htdocs\core\modules\project\doc\doc_generic_project_odt.modules.php:356 PhanTypeArraySuspicious Suspicious array access to $tasktime of type object
htdocs\core\modules\project\doc\doc_generic_project_odt.modules.php:357 PhanTypeArraySuspicious Suspicious array access to $tasktime of type object
htdocs\core\modules\project\doc\doc_generic_project_odt.modules.php:358 PhanTypeArraySuspicious Suspicious array access to $tasktime of type object
htdocs\core\modules\project\doc\doc_generic_project_odt.modules.php:359 PhanTypeArraySuspicious Suspicious array access to $tasktime of type object
htdocs\core\modules\project\doc\doc_generic_project_odt.modules.php:360 PhanTypeArraySuspicious Suspicious array access to $tasktime of type object
htdocs\core\modules\project\doc\doc_generic_project_odt.modules.php:361 PhanTypeArraySuspicious Suspicious array access to $tasktime of type object
htdocs\core\modules\project\doc\doc_generic_project_odt.modules.php:362 PhanTypeArraySuspicious Suspicious array access to $tasktime of type object
htdocs\core\modules\project\doc\doc_generic_project_odt.modules.php:363 PhanTypeArraySuspicious Suspicious array access to $tasktime of type object
htdocs\core\modules\project\task\doc\doc_generic_task_odt.modules.php:321 PhanTypeArraySuspicious Suspicious array access to $tasktime of type object
htdocs\core\modules\project\task\doc\doc_generic_task_odt.modules.php:322 PhanTypeArraySuspicious Suspicious array access to $tasktime of type object
htdocs\core\modules\project\task\doc\doc_generic_task_odt.modules.php:323 PhanTypeArraySuspicious Suspicious array access to $tasktime of type object
htdocs\core\modules\project\task\doc\doc_generic_task_odt.modules.php:324 PhanTypeArraySuspicious Suspicious array access to $tasktime of type object
htdocs\core\modules\project\task\doc\doc_generic_task_odt.modules.php:325 PhanTypeArraySuspicious Suspicious array access to $tasktime of type object
htdocs\core\modules\project\task\doc\doc_generic_task_odt.modules.php:326 PhanTypeArraySuspicious Suspicious array access to $tasktime of type object
htdocs\core\modules\project\task\doc\doc_generic_task_odt.modules.php:327 PhanTypeArraySuspicious Suspicious array access to $tasktime of type object
htdocs\core\modules\project\task\doc\doc_generic_task_odt.modules.php:328 PhanTypeArraySuspicious Suspicious array access to $tasktime of type object
htdocs\emailcollector\lib\emailcollector.lib.php:127 PhanTypeArraySuspicious Suspicious array access to $parts of type object|true
htdocs\webservices\server_actioncomm.php:457 PhanTypeArraySuspicious Suspicious array access to $actioncomm of type \ActionComm
htdocs\webservices\server_actioncomm.php:458 PhanTypeArraySuspicious Suspicious array access to $actioncomm of type \ActionComm
htdocs\webservices\server_actioncomm.php:459 PhanTypeArraySuspicious Suspicious array access to $actioncomm of type \ActionComm
htdocs\webservices\server_actioncomm.php:460 PhanTypeArraySuspicious Suspicious array access to $actioncomm of type \ActionComm
htdocs\webservices\server_actioncomm.php:461 PhanTypeArraySuspicious Suspicious array access to $actioncomm of type \ActionComm
htdocs\webservices\server_actioncomm.php:462 PhanTypeArraySuspicious Suspicious array access to $actioncomm of type \ActionComm
htdocs\webservices\server_actioncomm.php:463 PhanTypeArraySuspicious Suspicious array access to $actioncomm of type \ActionComm
htdocs\webservices\server_actioncomm.php:464 PhanTypeArraySuspicious Suspicious array access to $actioncomm of type \ActionComm
htdocs\webservices\server_actioncomm.php:465 PhanTypeArraySuspicious Suspicious array access to $actioncomm of type \ActionComm
htdocs\webservices\server_actioncomm.php:466 PhanTypeArraySuspicious Suspicious array access to $actioncomm of type \ActionComm
htdocs\webservices\server_actioncomm.php:467 PhanTypeArraySuspicious Suspicious array access to $actioncomm of type \ActionComm
htdocs\webservices\server_actioncomm.php:468 PhanTypeArraySuspicious Suspicious array access to $actioncomm of type \ActionComm
htdocs\webservices\server_actioncomm.php:469 PhanTypeArraySuspicious Suspicious array access to $actioncomm of type \ActionComm
htdocs\webservices\server_actioncomm.php:470 PhanTypeArraySuspicious Suspicious array access to $actioncomm of type \ActionComm
htdocs\webservices\server_actioncomm.php:471 PhanTypeArraySuspicious Suspicious array access to $actioncomm of type \ActionComm
htdocs\webservices\server_actioncomm.php:482 PhanTypeArraySuspicious Suspicious array access to $actioncomm of type \ActionComm
htdocs\webservices\server_actioncomm.php:537 PhanTypeArraySuspicious Suspicious array access to $actioncomm of type \ActionComm
htdocs\webservices\server_actioncomm.php:547 PhanTypeArraySuspicious Suspicious array access to $actioncomm of type \ActionComm
htdocs\webservices\server_actioncomm.php:552 PhanTypeArraySuspicious Suspicious array access to $actioncomm of type \ActionComm
htdocs\webservices\server_actioncomm.php:553 PhanTypeArraySuspicious Suspicious array access to $actioncomm of type \ActionComm
htdocs\webservices\server_actioncomm.php:554 PhanTypeArraySuspicious Suspicious array access to $actioncomm of type \ActionComm
htdocs\webservices\server_actioncomm.php:555 PhanTypeArraySuspicious Suspicious array access to $actioncomm of type \ActionComm
htdocs\webservices\server_actioncomm.php:556 PhanTypeArraySuspicious Suspicious array access to $actioncomm of type \ActionComm
htdocs\webservices\server_actioncomm.php:557 PhanTypeArraySuspicious Suspicious array access to $actioncomm of type \ActionComm
htdocs\webservices\server_actioncomm.php:558 PhanTypeArraySuspicious Suspicious array access to $actioncomm of type \ActionComm
htdocs\webservices\server_actioncomm.php:559 PhanTypeArraySuspicious Suspicious array access to $actioncomm of type \ActionComm
htdocs\webservices\server_actioncomm.php:560 PhanTypeArraySuspicious Suspicious array access to $actioncomm of type \ActionComm
htdocs\webservices\server_actioncomm.php:561 PhanTypeArraySuspicious Suspicious array access to $actioncomm of type \ActionComm
htdocs\webservices\server_actioncomm.php:562 PhanTypeArraySuspicious Suspicious array access to $actioncomm of type \ActionComm
htdocs\webservices\server_actioncomm.php:563 PhanTypeArraySuspicious Suspicious array access to $actioncomm of type \ActionComm
htdocs\webservices\server_actioncomm.php:564 PhanTypeArraySuspicious Suspicious array access to $actioncomm of type \ActionComm
htdocs\webservices\server_actioncomm.php:565 PhanTypeArraySuspicious Suspicious array access to $actioncomm of type \ActionComm
htdocs\webservices\server_actioncomm.php:566 PhanTypeArraySuspicious Suspicious array access to $actioncomm of type \ActionComm
htdocs\webservices\server_actioncomm.php:577 PhanTypeArraySuspicious Suspicious array access to $actioncomm of type \ActionComm
htdocs\webservices\server_actioncomm.php:603 PhanTypeArraySuspicious Suspicious array access to $actioncomm of type \ActionComm
htdocs\webservices\server_contact.php:401 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:410 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:411 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:412 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:413 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:414 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:415 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:416 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:417 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:418 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:419 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:420 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:421 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:422 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:423 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:424 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:425 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:426 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:427 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:428 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:429 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:430 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:431 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:432 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:433 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:434 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:435 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:436 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:437 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:438 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:439 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:440 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:441 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:452 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:638 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:644 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:656 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:661 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:663 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:664 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:666 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:667 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:668 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:670 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:671 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:672 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:674 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:677 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:678 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:679 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:680 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:681 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:683 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:684 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:686 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:697 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:703 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_contact.php:723 PhanTypeArraySuspicious Suspicious array access to $contact of type \Contact
htdocs\webservices\server_invoice.php:754 PhanTypeArraySuspicious Suspicious array access to $invoice of type \Facture
htdocs\webservices\server_invoice.php:755 PhanTypeArraySuspicious Suspicious array access to $invoice of type \Facture
htdocs\webservices\server_invoice.php:769 PhanTypeArraySuspicious Suspicious array access to $invoice of type \Facture
htdocs\webservices\server_invoice.php:779 PhanTypeArraySuspicious Suspicious array access to $invoice of type \Facture
htdocs\webservices\server_invoice.php:787 PhanTypeArraySuspicious Suspicious array access to $invoice of type \Facture
htdocs\webservices\server_invoice.php:790 PhanTypeArraySuspicious Suspicious array access to $invoice of type \Facture
htdocs\webservices\server_invoice.php:799 PhanTypeArraySuspicious Suspicious array access to $invoice of type \Facture
htdocs\webservices\server_invoice.php:800 PhanTypeArraySuspicious Suspicious array access to $invoice of type \Facture
htdocs\webservices\server_invoice.php:802 PhanTypeArraySuspicious Suspicious array access to $invoice of type \Facture
htdocs\webservices\server_invoice.php:803 PhanTypeArraySuspicious Suspicious array access to $invoice of type \Facture
htdocs\webservices\server_invoice.php:824 PhanTypeArraySuspicious Suspicious array access to $invoice of type \Facture
htdocs\webservices\server_productorservice.php:524 PhanTypeArraySuspicious Suspicious array access to $product of type \Product
htdocs\webservices\server_productorservice.php:525 PhanTypeArraySuspicious Suspicious array access to $product of type \Product
htdocs\webservices\server_productorservice.php:526 PhanTypeArraySuspicious Suspicious array access to $product of type \Product
htdocs\webservices\server_productorservice.php:690 PhanTypeArraySuspicious Suspicious array access to $product of type \Product
htdocs\webservices\server_productorservice.php:691 PhanTypeArraySuspicious Suspicious array access to $product of type \Product
htdocs\webservices\server_thirdparty.php:457 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:468 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:469 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:470 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:471 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:472 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:473 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:474 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:475 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:476 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:477 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:479 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:480 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:481 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:482 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:483 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:485 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:486 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:487 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:489 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:492 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:493 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:494 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:495 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:496 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:497 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:498 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:499 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:500 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:501 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:503 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:505 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:506 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:507 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:509 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:510 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:522 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:531 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:532 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:543 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:544 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:589 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:601 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:606 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:607 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:608 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:609 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:610 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:611 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:612 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:613 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:614 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:615 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:617 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:618 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:619 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:620 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:621 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:623 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:624 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:625 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:627 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:630 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:631 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:632 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:633 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:634 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:635 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:636 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:637 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:638 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:639 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:641 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:643 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:644 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:645 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:647 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:659 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:666 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
htdocs\webservices\server_thirdparty.php:686 PhanTypeArraySuspicious Suspicious array access to $thirdparty of type \Societe
```

**Comment by mdeweerd** [comment]
Do you think you can accept this or should I delete it?

**Comment by eldy** [comment]
The warning, for example first one: 
htdocs\core\modules\project\doc\doc_generic_project_odt.modules.php:352 PhanTypeArraySuspicious Suspicious array access to $tasktime of type object
was returned because the declaration of type field was wrong in doc. 
`@param  object			$tasktime			times object`
should be
`@param  array			$tasktime			times object`
Now this has been fixed, warning should have disappeared.

**Comment by mdeweerd** [comment]
I confirm that the only remaining PhanTypeArraySuspicious cases are the more specific PhanTypeArraySuspiciousNull and PhanTypeArraySuspiciousNullabal notifications.

**Comment by eldy** [comment]
I prefer to have a fix in code when we do an instance property access like an array, instead of fixing this by extending all classes into ArrayObject

**Pull request closed without merging**